docs-src/clusters/how-to-migrate-visibility-database.md

```diff
@@ -0,0 +1,119 @@
+---
+id: how-to-migrate-visibility-database
+title: How to migrate Visibility database
+sidebar_label: Migrating Visibility database
+description: To migrate your Visibility database to another database, set up a secondary Visibility to enable Dual Visibility, and update the dynamic configuration in your Cluster to update the Visibility store read and write operations.
+tags:
+  - operation-guide
+  - filtered-lists
+  - visibility
+ssdi:
+  - Supported from Temporal Server v1.21 onwards.
+---
+
+To migrate your Visibility database, [set up a secondary Visibility](/cluster-deployment-guide#set-up-secondary-visibility) to enable [Dual Visibility](/concepts/what-is-dual-visibility), and update the dynamic configuration in your Cluster to update the Visibility store read and write operations.
+
+Dual Visibility setup is optional but useful in gradually migrating your Visibility data to another database.
+
+Before you begin, verify [supported databases and versions](/cluster-deployment-guide#supported-databases) for a Visibility store.
+
+The following steps describe how to migrate your Visibility database with examples.
+
+#### Set up secondary Visibility store
+
+1. In your Cluster configuration, [add a secondary Visibility store](/references/configuration#secondaryvisibilitystore) to your Visibility setup under the Persistence configuration.
+
+   Example: To migrate from Cassandra to Elasticsearch, add Elasticsearch as your secondary database, and set it up. For details, see [secondary Visibility database schema and setup](/cluster-deployment-guide#set-up-secondary-visibility).
+
+   ```yaml
+   persistence:
+   visibilityStore: cass-visibility
+   secondaryVisibilityStore: es-visibility
+   datastores:
+       cass-visibility:
+       cassandra:
+           hosts: "127.0.0.1"
+           keyspace: "temporal_visibility"
+       es-visibility:
+       elasticsearch:
+           version: "v7"
+           logLevel: "error"
+           url:
+           scheme: "http"
+           host: "127.0.0.1:9200"
+           indices:
+           visibility: temporal_visibility_v1_dev
+           closeIdleConnectionsInterval: 15s
+   ```
```

Comment:
Should we add that they need to restart all services of the Temporal Server (history, workers, and frontend)? Maybe it's implied since it's changing the cluster config.

Reply:
yeah; I think best to add it in. This will become clearer once I start building out the cluster deployment guide.